src/core/lib/iomgr/tcp_windows.cc

```diff
@@ -100,29 +100,29 @@ typedef struct grpc_tcp {
   /* This is our C++ class derivation emulation. */
   grpc_endpoint base;
   /* The one socket this endpoint is using. */
-  grpc_winsocket* socket;
+  grpc_winsocket* socket = nullptr;
```

Comment:
qq: how come the pointers need to be explicitly initialized to nullptr, but other members (such as base, refcount, on_read, etc) aren't explicitly initialized (and will their members get properly initialized?)? Maybe I'm missing something, but it seems like an inconsistency?

Reply:
Good point. My idea was to initialize simple types like pointers and integers and avoid the struct types. But it might be a good idea to initialize those with {}. Since that, to my knownledge, initializes it's members to zero.

Reply:
@yashykt  would know what's the best thing to do here.

Reply:
This still hasn't been addressed.

Reply:
To avoid delaying this any further, let's merge this after adding a TODO here to C++ify the depedencies, i.e., add constructors so that we know that things are properly initialized.

Reply:
I rebased and added the TODO.